router/batchrouter/asyncdestinationmanager/yandexmetrica/yandexmetrica.go

```diff
@@ -0,0 +1,382 @@
+package yandexmetrica
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/samber/lo"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/yandexmetrica/augmenter"
+	oauthv2 "github.com/rudderlabs/rudder-server/services/oauth/v2"
+	oauthv2common "github.com/rudderlabs/rudder-server/services/oauth/v2/common"
+	cntx "github.com/rudderlabs/rudder-server/services/oauth/v2/context"
+	oauthv2httpclient "github.com/rudderlabs/rudder-server/services/oauth/v2/http"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var (
+	json        = jsoniter.ConfigCompatibleWithStandardLibrary
+	idClientMap = map[string]string{
+		"ClientId": "CLIENT_ID",
+		"Yclid":    "YCLID",
+		"UserId":   "USER_ID",
+	}
+)
+
+type YandexMetricaMessageBody struct {
```

Comment:
Do we need to export this type? I don't see it used outside of this package

```suggestion
type yandexMetricaMessageBody struct {
```

Reply:
Updated